pkg/pillar/types/global.go

```diff
@@ -426,66 +424,43 @@ func EnforceGlobalConfigMinimums(newgc GlobalConfig) GlobalConfig {
 
 // Agents which wait for GlobalConfig initialized should call this
 // on startup to make sure we have a GlobalConfig file.
-func EnsureGCFile() {
-	if _, err := os.Stat(globalConfigDir); err != nil {
-		log.Infof("Create %s\n", globalConfigDir)
-		if err := os.MkdirAll(globalConfigDir, 0700); err != nil {
-			log.Fatal(err)
+func EnsureGCFile(pub *pubsub.Publication) {
```

Comment:
Is this the only place where Types is dependent on PubSub? I am wondering of this should move to some Utils??

Reply:
types depend on pubsub in lots of places.

Reply:
Turns out there were only two so I moved that code into the utils package so we don't need duplication vs circular dependencies.